packages/babel-parser/test/fixtures/es2015/yield/parameter-default-inside-arrow-inside-generator-5/output.json

```diff
@@ -2,8 +2,9 @@
   "type": "File",
   "start":0,"end":47,"loc":{"start":{"line":1,"column":0},"end":{"line":3,"column":1}},
   "errors": [
-    "SyntaxError: Can not use 'yield' as identifier inside a generator (2:8)",
-    "SyntaxError: Binding invalid left-hand side in function parameter list (2:8)"
+    "SyntaxError: yield is not allowed in generator parameters (2:8)",
+    "SyntaxError: Binding invalid left-hand side in function parameter list (2:8)",
+    "SyntaxError: yield is not allowed in generator parameters (2:8)"
```

Comment:
This is, in a way, the side effect of recording errors in every ancestry scope (until expression scope).

https://github.com/babel/babel/blob/d51aa6d76177b544590cdfe3868f9f4d33d8813d/packages/babel-parser/test/fixtures/es2015/yield/parameter-default-inside-arrow-inside-generator-5/input.js#L1-L3

In the error recovery mode, since the error is recorded in both `(yield)` and (x = (yield) => {})`, it will be thrown twice although they are both caused by `yield`. We could cleanup ancestry errors in `validateAsPattern` -- but we are not aware whether certain errors are caused by the same `yield` identifier`.

Note that we already throw 2 different errors for this case, which is not ideal.